src/test/java/co/rsk/federate/watcher/FederationWatcherTest.java

```diff
@@ -79,248 +76,317 @@ void whenFederationWatcherIsSetUp_shouldAddListener() throws Exception {
     }
 
     @Test
-    void whenNoActiveFederation_shouldTriggerActiveFederationChange() throws Exception {
+    void onBestBlock_whenNoActiveFederationAndProposedFederationExists_shouldTriggerActiveAndProposedFederationChange() throws Exception {
         // Arrange
-        var rskListener = setupAndGetRskListener(null, null);
-        var activeCalls = new AtomicInteger(0);
-        var retiringCalls = new AtomicInteger(0);
-
+        var rskListener = setupAndGetRskListener(null, null, null);
+        when(federationProvider.getProposedFederationAddress()).thenReturn(Optional.of(SECOND_FEDERATION.getAddress()));
+        when(federationProvider.getProposedFederation()).thenReturn(Optional.of(SECOND_FEDERATION));
         when(federationProvider.getActiveFederationAddress()).thenReturn(FIRST_FEDERATION.getAddress());
         when(federationProvider.getActiveFederation()).thenReturn(FIRST_FEDERATION);
         when(federationProvider.getRetiringFederationAddress()).thenReturn(Optional.empty());
 
+        federationWatcher.start(federationProvider, federationWatcherListener);
+
         // Act
-        federationWatcher.start(federationProvider, new FederationWatcherListener() {
-            @Override
-            public void onActiveFederationChange(Federation newFederation) {
-                assertEquals(FIRST_FEDERATION, newFederation);
-                activeCalls.incrementAndGet();
-            }
-
-            @Override
-            public void onRetiringFederationChange(Federation newFederation) {
-                retiringCalls.incrementAndGet();
-            }
-        });
         rskListener.onBestBlock(null, null);
 
         // Assert
-        assertEquals(1, activeCalls.get());
-        assertEquals(0, retiringCalls.get());
+        verify(federationProvider).getProposedFederationAddress();
+        verify(federationProvider).getProposedFederation();
+        verify(federationWatcherListener).onProposedFederationChange(SECOND_FEDERATION);
+
         verify(federationProvider).getActiveFederationAddress();
-        verify(federationProvider).getRetiringFederationAddress();
+        verify(federationProvider).getActiveFederation();
+        verify(federationWatcherListener).onActiveFederationChange(FIRST_FEDERATION);
+
         verify(federationProvider).getActiveFederation();
```

Comment:
```suggestion
        verify(federationProvider).getActiveFederation();
        verify(federationWatcherListener).onActiveFederationChange(FIRST_FEDERATION);
```
it is duplicated right?

Reply:
Thanks, updated